engine/src/cmd/collide_map.cpp

```diff
@@ -161,8 +160,8 @@ void CollideArray::flatten() {
             collideUpdate(*tmp, index);
         }
 
-        std::list<CollidableBackref>::iterator listend = toflattenhints[i].end();
-        for (std::list<CollidableBackref>::iterator j = toflattenhints[i].begin();
+        auto listend = toflattenhints[i].end();
```

Comment:
I'm fine with type inference where it is trivial. Example from other languages is something like

`let someString = String('blah blah')`

However, I'm less comfortable with replacing _std::list<CollidableBackref>::iterator_ with _auto_. I prefer to clearly see my C++ abominations.

Reply:
Fair point LOL.

I was going by a CLion suggestion that all iterator variables should use `auto`. Sounds like we need to discuss that further though.

Reply:
Started a discussion for this: https://github.com/orgs/vegastrike/discussions/800

Reply:
I'm okay with `auto` for the loops with iterators because (a) the loops should be fairly short, (b) the iterator notation (`namespace::type::iterator`) can get quite long and lengthy and doesn't serve much purpose, and (c) the iterator type can change somewhat so it's helpful.

So this is actually one of the few places I think `auto` is okay. That said, I already mentioned that the `end` check needs to be done in the loop since `end` could  change or become invalid during the loop.